src/Table/index.js

```diff
@@ -46,18 +44,24 @@ class Table extends PureComponent {
 
   componentDidMount() {
     const { isScrollable } = this.props;
+    this.checkDataDepth();
+
     if (isScrollable) {
       this.setFirstCellWidth();
       addEventListener('resize', this.onResize);
       this.containerRef.current.addEventListener('scroll', this.onScroll);
     }
   }
 
-  componentDidUpdate({ colsDef: prevColsDef }) {
-    const { colsDef } = this.props;
+  componentDidUpdate({ colsDef: prevColsDef, data: prevData }) {
+    const { colsDef, data } = this.props;
+
     if (prevColsDef !== colsDef) {
```

Comment:
@anglol (since git blame seemed to point the author to you :d) I took the same logic l.62 but actually that will always be false (objects and arrays are never equal). Did not take time to dig that up but maybe someone has an idea, besides doing a deep check?

Reply:
ok, I will take a look